backend/src/data/repositories/course/course.repository.ts

```diff
@@ -15,10 +18,24 @@ class Course {
     this.#CourseModel = CourseModel;
   }
 
-  public getAll(): Promise<CourseGetResponseDto[]> {
+  public getAll(opts?: {
+    filtering: CourseFilteringDto;
+  }): Promise<CourseGetResponseDto[]> {
+    const { title = '' } = opts?.filtering ?? {};
+
     return this.#CourseModel
       .query()
       .withGraphJoined('vendor')
+      .where((QueryBuilder) => {
+        if (!title) {
+          return;
+        }
+        QueryBuilder.where(
+          raw('LOWER(??)', 'title'),
+          'like',
+          `%${title.toLowerCase()}%`,
+        );
```

Comment:
You can simply use case-insensitive `ilike`, instead of lower-casing everything
```suggestion
      .where((builder) => {
        if (!title) {
          return;
        }
        builder.where(
          'title', 'ilike', `%${title}%`
        );
```